src/server/controllers/UserService.ts

```diff
@@ -1,115 +1,94 @@
-            }).catch(err => {
-                logger.error('Create Failed with: ', err);
+        return Users.sync()
+            .then(() => Users.create(user))
+            .then((newUser) => newUser.toJSON())
+            .catch((err) => {
+                logger.error('create failed with: ', err);
                 return undefined;
             });
     }
 
-    public static update(email: string, user: any): Promise<IUserModel | undefined> {
-        // TODO: user validation
-
-        // build 'SET' query
-        let updateStr = '';
-        for (const key of ['email', 'name', 'userClass', 'hash', 'salt']) {
-            if (key in user) {
-                updateStr += `${key}='${user[key]}', `;
-            }
-        }
-
-        // Remove the extra comma and space
-        if (updateStr.length !== 0) {
-            updateStr = updateStr.substring(0, updateStr.length - 2);
-        } else {
-            throw new Error('user should have valid attributes to be updated');
-        }
+    public static update(id: number, user: any): Promise<IUserModel | undefined> {
 
-        return query(`UPDATE users
-                        SET ${updateStr}
-                        WHERE email='${email}'`)
-            .then((res) => {
-                logger.info('Updating user: ', email);
-                return user;
+        return Users.sync()
+            .then(() => Users.update(user, { where: { user_id: id }, returning: true }))
+            .then(val => {
+                const [num, users] = val;
+                if (num === 0) {
+                    logger.error('update id matched no existing user');
+                    return undefined;
+                } else if (num > 1) {
+                    logger.error('updated too many users');
+                    return undefined;
+                }
+                return users[0]!.toJSON();
             }).catch((err) => {
-                logger.error('Updated Failed with: ', err);
+                logger.error('update failed with: ', err);
                 return undefined;
             });
     }
 
-    public static delete(email: string): Promise<void> {
-        return query(`DELETE FROM users WHERE email='${email}'`)
-            .then((res) => {
-                logger.info('Deleting user: ', email);
-            }).catch((err) => {
-                logger.error('Delete failed with: ', err);
+    public static delete(id: number): Promise<void> {
+        return Users.sync()
+            .then(() => Users.destroy({ where: { user_id: id } }))
+            .then((num) => {
+                if (num === 0) {
+                    logger.error('delete deleted no rows');
+                    throw new Error('No rows deleted');
+                } else if (num > 1) {
+                    logger.error('delete deleted more than one row');
+                    throw new Error('More than one row deleted');
+                }
             });
     }
 
     public static isEmpty(): Promise<boolean> {
```

Comment:
isEmpty is called on user registration so that the first user created is an admin, presumably whoever deployed the instance

idk this is entirely matter of preference. Should we instead have the default admin/user be defined in environment variables?

Reply:
that makes sense then. I don't mind doing it this way